packages/connections/src/components/connecting/connecting-animation.tsx

```diff
@@ -0,0 +1,154 @@
+  marginTop: spacing[3],
+  textAlign: 'center',
+});
+
+const animationSvgStyles = css({
+  width: 70,
+  height: 'auto',
+});
+
+const shadowStyles = css({
+  fill: '#136149',
+  opacity: 0.12,
+});
+
+const ringCircleStyles = css({
+  stroke: '#136149',
+  strokeLinecap: 'round',
+  strokeLinejoin: 'round',
+  fill: 'none',
+});
+
+const ringShadowStyles = css(ringCircleStyles, {
+  opacity: 0.12,
+});
+
+const innerCircleStyles = css({
+  fill: '#fef7e3',
+  opacity: 0.85,
+});
+
+const outerCircleStyles = css({
+  fill: '#f7a76f',
+});
+
+const redArrowStyles = css({
+  fill: '#FF5516',
+});
+
+const arrowStyles = css({
+  fill: 'rgba(9, 128, 76, 0.3)',
+});
+
+// This function returns the speed at which the needle shoots off in
+// a direction. The farther from 0 the number, the farther/faster it goes.
+const getNewRotationVelocity = () => {
+  return (
+    (Math.PI / (170 + Math.random() * 100)) * (Math.random() > 0.5 ? 1 : -1)
+  );
+};
+
+// How fast the needle returns to the center mark.
+const rotationAcceleration = Math.PI / 90000;
+// Closer to 0 the more friction/slowdown overtime there is (1 is no friction).
+const friction = 0.974;
+
+/**
+ * Animated compass shown when attempting to connect.
+ */
+function ConnectingAnimation(): React.ReactElement {
+  const requestAnimationRef =
+    useRef<ReturnType<typeof window.requestAnimationFrame>>();
+  const lastFrame = useRef<number>(Date.now());
+  const currentRotation = useRef<number>(0);
+  const rotationVelocity = useRef<number>(getNewRotationVelocity());
+
+  function updateAnimation() {
+    if (Date.now() - lastFrame.current > 20) {
+      // When the user returns from an unfocused view we disregard
+      // that last frame time for a frame.
+      lastFrame.current = Date.now();
+    }
+
+    const deltaTime = Date.now() - lastFrame.current;
+
+    const arrow1 = document.getElementById('connectingArrow1');
+    const rotation = currentRotation.current * (180 / Math.PI);
+    arrow1?.setAttribute('transform', `rotate(${rotation}, 24.39, 39.2)`);
+    const arrow2 = document.getElementById('connectingArrow2');
+    arrow2?.setAttribute('transform', `rotate(${rotation}, 24.39, 39.2)`);
+
+    currentRotation.current += rotationVelocity.current * deltaTime;
+    rotationVelocity.current +=
+      rotationAcceleration * (currentRotation.current > 0 ? -1 : 1) * deltaTime;
+    rotationVelocity.current *= friction;
+
+    if (
+      Math.abs(rotationVelocity.current) < Math.PI / 1100 &&
+      Math.abs(currentRotation.current) < Math.PI / 1100
+    ) {
+      // When the Compass hands are settled we apply a force so
+      // it starts to rotate again.
+      rotationVelocity.current = getNewRotationVelocity();
+    }
+
+    lastFrame.current = Date.now();
+
+    requestAnimationRef.current = window.requestAnimationFrame(updateAnimation);
+  }
+
+  useEffect(() => {
+    requestAnimationRef.current = window.requestAnimationFrame(updateAnimation);
```

Comment:
If it's alright I think it makes sense to be explicit here about using `window. requestAnimationFrame` and assigning it since that's involved in the cleanup we do. Although it wouldn't be an issue I think having that in code makes it a bit clearer where we're setting it and cleaning it up. We could change this to `useLayoutEffect` if we want to start the calling in the rendering cycle, but I think we probably don't need to.

Reply:
Good point about having `requestAnimationRef.current` set immediately in the effect on the first one, makes sense 👍